src/Config.php

```diff
@@ -198,6 +198,23 @@ class Config
      */
     private static $executablePaths = [];
 
+    /**
+     * A list of valid generators.
+     *
+     * - Keys: lowercase version of the generator name.
+     * - Values: name of the generator PHP class.
+     *
+     * Note: once support for PHP < 5.6 is dropped, this property should be refactored into a class
+     * constant.
+     *
+     * @var array<string, string>
+     */
+    private static $validGenerators = [
```

Comment:
Is there a reason for making this property `static` ?

Reply:
I made this property `static` because I see its value as common for all instances of this class and not something that will change depending on the instance of the class. Do you see other reasons not to make it `static`?

Anyway, I had to change it to an object property to be able to manipulate its value when generating the error message (copying the value of the property before manipulating it seemed too much). See https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/771#discussion_r1893267997